icinga-app/icinga.cpp

```diff
@@ -606,34 +606,25 @@ static int Main()
 			errno = 0;
 			struct passwd *pw = getpwnam(user.CStr());
 
-			if (!pw) {
-				if (errno == 0) {
-					Log(LogCritical, "cli")
-						<< "Invalid user specified: " << user;
-					return EXIT_FAILURE;
-				} else {
-					Log(LogCritical, "cli")
-						<< "getpwnam() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-					return EXIT_FAILURE;
-				}
-			}
-
-			// also activate the additional groups the configured user is member of
-			if (getuid() != pw->pw_uid) {
-				if (!vm.count("reload-internal") && initgroups(user.CStr(), pw->pw_gid) < 0) {
-					Log(LogCritical, "cli")
-						<< "initgroups() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-					Log(LogCritical, "cli")
-						<< "Please re-run this command as a privileged user or using the \"" << user << "\" account.";
-					return EXIT_FAILURE;
-				}
+			// only respect groups if there exists a passwd entry for the current user
```

Comment:
This comment does not represent what actually happens in the following block.

`pw` is populated by `getpwnam(3)` on `Configuration::RunAsUser`, defaulting to `icinga2` or `nagios` (on Debian), but can be overwritten both during compile and runtime via `ICINGA2_USER`. Thus, if such a user is present in the user database, this `if` block is being accessed.

I only see one group-related code within, and this is the `initgroups(3)` call based on the user name and `pw->pw_gid`. Does this fail for you?